### PR TITLE
[ISSUE-856] report non default namespace drive event

### LINKED ISSUE #856: When csi-baremetal is deployed on non-default namespace, the CSI baremetal application generates events in the default namespace
**Describe the bug**
When csi-baremetal is deployed in non-default namespace and when the disk fails, the csi-baremetal application generates events in the default namespace.

**Environment (please complete the following information):**
 - Hardware: Any
 - OS: Any
 - Version: 1.1.0

**To Reproduce**
1. Deploy csi-baremetal in deployed in non-default namespace
2. Disk failure
3. kubectl get events -A to check the DriveHealthFailure  event on default namespace.

**Expected behavior**
When csi-baremetal is deployed in non-default namespace and when the disk fails, the csi-baremetal application generates events in the non-default namespace.

**Screenshots**
[root@centos76 ~]# kubectl get events -A | grep 40S0A0KRF1QF
default 6s Error DriveHealthFailure drive/09275394-45ed-426f-a735-0aaaf67ab326 Drive health is: BAD, previous state: GOOD. Drive Details: SN='40S0A0KRF1QF', Model='ATA TOSHIBA MG06ACA8', Type='HDD', Size='8001000000000', Firmware='GA08', Slot='21', NodeName='master0.ocp4.carnelian.com'
[root@centos76 ~]#
**Additional context**

### PR DESCRIPTION
**Pull request by yanxiucai**
Signed-off-by: yanxiucai <Yanxiu_Cai@dell.com>

### Resolves #856 

Update the code in https://github.com/dell/csi-baremetal/blob/master/pkg/events/recorder/recorder.go function makeEvent() to generate events for drives in the same namespace where CSI is installed.
## PR checklist
- [x] Add link to the issue
- [x] Choose Project
- [x] Choose PR label
- [x] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
[test.txt](https://github.com/dell/csi-baremetal/files/8844036/test.txt)

### CODE REVIEW

**Comment by mishoyama** [comment]
@yanxiucai please fix `codecov/patch` check

File: pkg/events/recorder/recorder.go [hunk 1]
```diff
@@ -67,7 +68,12 @@ func (sr *SimpleRecorder) makeEvent(ref *v1.ObjectReference, labels map[string]s
 	t := metav1.NewTime(now)
 	namespace := ref.Namespace
 	if namespace == "" {
-		namespace = metav1.NamespaceDefault
+		if key := os.Getenv("NAMESPACE"); key != "" {
```
Comment (mishoyama) [inline pkg/events/recorder/recorder.go:71]: CSI generates events for cluster scoped CRs like `drives` and namespaced CRs like `volumes`.  How will that work when CSI in installed in non-default namespace, but volume CR is in default?

Comment (yanxiucai) [reply pkg/events/recorder/recorder.go:71]: I think we need to add a check here, it is for drives only.

File: pkg/events/recorder/recorder.go [hunk 2]
```diff
@@ -67,7 +68,12 @@ func (sr *SimpleRecorder) makeEvent(ref *v1.ObjectReference, labels map[string]s
...
+			namespace = os.Getenv("NAMESPACE")
```
Comment (libzhang) [inline pkg/events/recorder/recorder.go:72]: no need to use os.Getenv to retrieve NAMESPACE again,  use 
```
namespace = key  
```
is enough.

Comment (yanxiucai) [reply pkg/events/recorder/recorder.go:72]: Yes.

File: pkg/events/recorder/recorder.go (hunk 1 above)
Comment (yanxiucai) [reply pkg/events/recorder/recorder.go:71]: updated

Comment (mishoyama) [reply pkg/events/recorder/recorder.go:71]: Sorry, my comment was wrong. For cluster scoped CRs namespace will be empty, for namespaced - non-empty. @safronovD am I correct?

Comment (yanxiucai) [reply pkg/events/recorder/recorder.go:71]: ref.Nmespace here is empty or "default" for the namespaced CR while it is in default namespace?

Comment (safronovD) [reply pkg/events/recorder/recorder.go:71]: In my opinion, the first implementation was correct - `namespace == ""` means that resource is cluster-scoped, so we generate event for it in CSI NS and it doesn't matter is it `Drive` or something else

**Review by vgupta-mickey — COMMENTED** [review]
I am not 100% how to resolve this issue. However,  when you generate event on a cluster-scoped resource, the InvolvedObject.Name = cluster-scoped eresource and InvolvedObject.Namespace = "". 
The Event.ObjectMeta.Namespace = default.

**Comment by yanxiucai** [comment]
> However, when you generate event on a cluster-scoped resource, the InvolvedObject.Name = cluster-scoped eresource and InvolvedObject.Namespace = "".
> The Event.ObjectMeta.Namespace = default.

@vgupta-mickey In original implement, it always generate the event on cluster-scoped resource (drive) in default namespace but not in the namespace that CSI is installed.

**Comment by yanxiucai** [comment]
Close this PR as no code change is needed in CSI side.

**Pull request closed without merging**